File: 04_quicksort/javascript/04_recursive_max.js
function max(arr){
	if (arr.length == 2){
		return arr[0] > arr[1] ? arr[0] : arr[1];
	}
	var temp = arr[0];
	arr.splice(0,1)
	var sub_max = max(arr)
	return temp > sub_max ? temp : sub_max;
} 

console.log(max([1, 2, 3, 4]));

File: 04_quicksort/swift/04_recursive_max.swift
func max(list: [Int]) -> Int {
	if list.count == 2 {
		return list[0] > list[1] ? list[0] : list[1]
	}
	var newlist = list
	let temp = list[0]
	newlist.removeFirst()
	let sub_max = max(list: newlist)
	return temp > sub_max ? temp : sub_max
}

print(max(list: [1, 2, 3, 4]))
